dogm/demo/utils/include/precision_evaluator.h

```diff
@@ -23,20 +28,22 @@ class PrecisionEvaluator
 {
 public:
     PrecisionEvaluator(const SimulationData sim_data, const float resolution, const float grid_size);
+
+    void registerMetric(const std::string& name, std::shared_ptr<Metric> metric);
```

Comment:
You only want to have one instance of a Metric, right? Consider unique_ptr (maybe in conjunction with std::move()).

Reply:
std::move does the trick 👍 . I used unique_ptr before and that didn't work :smile: